nova/scheduler/utils.py

```diff
@@ -1483,3 +1484,31 @@ def get_aggregates_for_routed_subnet(
                 'Failed to find aggregate related to segment %s' % segment_id)
         return agg_info.aggregates
     return []
+
+
+def is_non_vmware_spec(spec_obj):
+    """Tests, if scheduler spec is specifiying a non-vmware hypervisor
+
+    That is SAP specific, as we run majorly on VMware, and target it
+    as the default
+
+    """
+    if 'flavor' in spec_obj:
+        if nova_utils.is_baremetal_flavor(spec_obj.flavor):
+            return True
+        extra_specs = spec_obj.flavor.extra_specs
+        hypervisor_type = extra_specs.get('capabilities:hypervisor_type')
```

Comment:
How does this work? I can't find the `capabilities:hypervisor_type` extra spec in any of our flavors. Or do you propose adding this to our flavors?

Reply:
Yes, you can target a hypervisor with the capabilities filter by setting that attribute. It also works by just setting i.e. `hypervisor_type=ironic`, but that is the deprecated way.

Reply:
To elaborate a bit more "capabilities:<attribute>" test any attribute of the HostState

Reply:
Missed to answer your original question: Setting it on the flavor is one of the options discussed, the other is setting it on the image. The code is intended to work with either approach.

Reply:
ah okay, so if merged, this wouldn't work out of the box, we'd have to set the image or flavor properties through the seeder. got it.

Reply:
we already set `vmware:hv_enabled` on our flavors. why not use that one?

Reply:
That is a parameter for the vmware hypervisor (as all parameters prefixed with `vmware:` are), and this one enabled hypervisor capability on that hypervisor. That parameters for a hypervisor are set doesn't mean that the instance with that flavor will end up on that hypervisor.

I am looking at the parameters which the scheduler uses to decide to schedule to a certain hypervisor (in [ComputeCapabilitiesFilter](https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/scheduler/filters/compute_capabilities_filter.py) )